Backend/sub_crates/language/src/tools/get.rs

```diff
@@ -10,17 +10,23 @@ impl Get for Dictionary {
         let lang_table = self.table.read().unwrap();
         let lang_index = language as usize;
         match lang_table.get(key) {
-            Some(translations) => match translations.get(lang_index) {
-                Some(value) => {
-                    if value.is_empty() {
-                        panic!("Value is empty for key {} and language {}", key, lang_index)
-                    } else {
-                        value.to_string()
-                    }
-                },
-                None => panic!("Key {} is not registered for language {}", key, lang_index),
+            Some(translations_vec) => {
+                let len = translations_vec.len();
+                if len <= lang_index {
```

Comment:
Verwende in Zeile 18 explizit die .get funktion, die gibt auch none für ein index aus, wofür es keinen key gibt.

Reply:
Vielleicht habe ich mir bei diesem (häslichen) Konstrukt was gedacht. Fällt mir jetzt aber nicht mehr ein... Morgen nochmal drüber nachdenken

Reply:
Ah, das war mein Handling für das Option<Option<>>. Mit unwrap_or siehts schöner aus.